test/checkChallenge.test.js

```diff
@@ -298,135 +352,182 @@ describe('Challenges', () => {
+      const notification = base_challenge({ ranked: false, user: { ranking: 26 }, handicap: -1 }); // "4k"
 
-      config.minhandicap = 0;
-      config.maxhandicap = 6;
+      config.fakerank    = 29; // "1k"
+      // "4k" - "1k" = 26 - 29 = 3 automatic handicap stones
+      config.minhandicap = 2;
+      config.maxhandicap = 4;
       
-      const result = conn.checkChallengeMinMax(notification);
+      const result = conn.checkChallengeHandicap(notification);
       
       assert.deepEqual(result, ({ reject: false }));
 
     });
 
-    it('accept handicap edge min (fakerank automatic handicap stone number estimation)', () => {
+    it('accept automatic handicap (-1) if fakerank is set and estimated handicap stones are edge max', () => {
 
-      const notification = base_challenge({ ranked: false, user: { ranking: 29 }, handicap: -1 }); // "1k"
+      const notification = base_challenge({ ranked: false, user: { ranking: 25 }, handicap: -1 }); // "5k"
 
       config.fakerank    = 29; // "1k"
-      // "1k" - "1k" = 29 - 29 = 0 automatic handicap stones
-      config.minhandicap =  0;
-      config.maxhandicap =  6;
+      // "5k" - "1k" = 25 - 29 = 4 automatic handicap stones
+      config.minhandicap = 2;
+      config.maxhandicap = 4;
       
-      const result = conn.checkChallengeMinMax(notification);
+      const result = conn.checkChallengeHandicap(notification);
       
       assert.deepEqual(result, ({ reject: false }));
 
     });
 
-    it('accept handicap between min and max', () => {
+    it('reject automatic handicap (-1) if fakerank is set but estimated handicap stones are too high', () => {
 
-      const notification = base_challenge({ ranked: false, handicap: 1 });
+      const notification = base_challenge({ ranked: false, user: { ranking: 30 }, handicap: -1 }); // "1d"
 
-      config.minhandicap = 0;
+      config.fakerank    = 20; // "10k"
+      // "10k" - "1d" = 30 - 20 = 10 automatic handicap stones
+      config.minhandicap = 2;
+      config.maxhandicap = 4;
+      
+      const result = conn.checkChallengeHandicap(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Maximum number of handicap stones is 4, please reduce the number of handicap stones.' }));
+
+    });
+
+  });
+
+  describe('Handicap', () => {
+
+    it('reject handicap too low (handicap games only)', () => {
+
+      const notification = base_challenge({ ranked: false, handicap: 0 });
+
+      config.noautohandicap = true;
+      config.minhandicap = 2;
       config.maxhandicap = 6;
       
-      const result = conn.checkChallengeMinMax(notification);
+      const result = conn.checkChallengeHandicap(notification);
       
-      assert.deepEqual(result, ({ reject: false }));
+      assert.deepEqual(result, ({ reject: true,   msg: 'Minimum number of handicap stones is 2, please increase the number of handicap stones (handicap games only).' }));
```

Comment:
What does `(handicap games only)` mean? I think we can remove it.

Alternatively we could write "This bot only plays handicap games.", but that might be difficult, as there is also an option for ranked/unranked to be different.

Reply:
no actually this would be easier, we pass argName to  function and get "" or "ranked " or unranked "
problem would be we lose the information of minimum handicap

so (handicap games only) was more of an extra information, i think you're right and we should delete it because it adds confusion, and min handicap info is enough in itself

Reply:
yes i agree, if max handicap is 0, "no handicap games" is very useful. However the reverse had me itching as, beside the poor wording, was also inaccurate.